### PR TITLE
Fix missing x y attrs sh

### PR DESCRIPTION
**Pull request by trey-stafford**

### CODE REVIEW

File: scripts/ancillary/fix-sh-xy-attrs.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+from seaice_ecdr.ancillary import get_ancillary_filepath
+
+
+def fix_xy_attrs(*, ancillary_ds: xr.Dataset) -> xr.Dataset:
+    ancillary_ds.x.attrs = dict(
+        standard_name="projection_x_coordinate",
+        long_name="x coordinate of projection",
+        units="meters",
+        axis="X",
+        valid_range=(ancillary_ds.x.min(), ancillary_ds.x.max()),
```
Comment (sc0tts) [inline scripts/ancillary/fix-sh-xy-attrs.py:19]: I think this valid range assignment isn't quite right.
There's a difference between the "valid_range" and the "actual_range".  

The latter refers to the values in the arrays -- i.e., what would be returned by .min() and .max() functions.

The valid_range should include the entire domain of the grid, and so should go from "x.min() - half-the-grid-cell-width" to "x.max() + half-the-grid-cell-width"

File: scripts/ancillary/fix-sh-xy-attrs.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    )
+
+    ancillary_ds.y.attrs = dict(
+        standard_name="projection_y_coordinate",
+        long_name="y coordinate of projection",
+        units="meters",
+        axis="Y",
+        valid_range=(ancillary_ds.y.min(), ancillary_ds.y.max()),
```
Comment (sc0tts) [inline scripts/ancillary/fix-sh-xy-attrs.py:27]: As above, valid_range != actual_range

**Review by sc0tts — CHANGES_REQUESTED** [review]
At some point, we probably need to have a single-source-of-truth for the geospatial netCDF variables/coordinates/dimensions.  

For now, I've been using the appropriate fields in NSIDC-0771 for the polar-stereo grids.

File: scripts/ancillary/fix-sh-xy-attrs.py (hunk 1 above)
Comment (trey-stafford) [reply scripts/ancillary/fix-sh-xy-attrs.py:19]: Good catch!

Comment (trey-stafford) [reply scripts/ancillary/fix-sh-xy-attrs.py]: 38c2115

**Pull request merged**